modules/auxiliary/scanner/etcd/open_key_scanner.rb

```diff
@@ -0,0 +1,72 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name'        => 'Etcd Keys API Information Gathering',
+      'Description' => %q{
+        This module queries the etcd API to recursively retrieve all of the stored
+        key value pairs.  Etcd by default does not utilize authentication.
+      },
+      'References'  => [
+          ['URL', 'https://elweb.co/the-security-footgun-in-etcd']
+        ],
+      'Author'      => [
+          'Giovanni Collazo <hello@gcollazo.com>', # discovery
+          'h00die' # msf module
+        ],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options([
+      Opt::RPORT(2379),
+      OptString.new('TARGETURI', [ true,  'URI of the vulnerable service', '/'])
+    ])
+  end
+
+  def run_host(target_host)
+    path = normalize_uri(target_uri.to_s, 'v2/keys/?recursive=true')
+
+    vprint_status("#{peer} - Collecting data through #{path}...")
+    res = send_request_raw({
+      'uri'    => path,
+      'method' => 'GET'
+    })
+
+    # parse the json if we got a good request back
+    if res and res.code == 200
```

Comment:
Minor: [prefer && to 'and'](https://github.com/bbatsov/ruby-style-guide#no-and-or-or)